### DISCUSSION TITLE
stall and/or flush counter

### DISCUSSION THREAD

**Discussion by lcaimi** [mortbopet/Ripes#173] [Q&A] [answered]
Is there any counter available to see the number of stalls or flushes during the code execution?

**Comment by mortbopet** [comment, marked as answer]
Hi @lcaimi 
Nope! no such counter is currently implemented. But i think it could be a good project for starting to get into Ripes development, or you could raise an issue as a feature request for it.
For inspiration, if looking at these two functions, one should be able to follow all the way from how the statistics are registerred inside the processor model, to them being displayed in the UI.
https://github.com/mortbopet/Ripes/blob/fe4b16ed657a59a1f94a986db33237605ad8084c/src/processors/interface/ripesprocessor.h#L208-L217

**Comment by lcaimi** [comment]
I saw that the StageInfo struct maintains the current state of each stage. I believe this information could be used to count stalls and flushes.

https://github.com/mortbopet/Ripes/blob/fe4b16ed657a59a1f94a986db33237605ad8084c/src/processors/interface/ripesprocessor.h#L19-L29

I am not able to implement the functionality. I will create an issue for future implementation.

Thanks
